docs/bundler/migration.md

```diff
@@ -0,0 +1,1107 @@
+- `--chunk-naming`
+- Renamed for consistency with `naming` in JS API
+
+---
+
+- `--color`
+- n/a
+- Always enabled
+
+---
+
+- `--drop`
+- n/a
+- Not supported
+
+---
+
+- `--entry-names`
+- `--entry-naming`
+- Renamed for consistency with `naming` in JS API
+
+---
+
+- `--footer`
+- n/a
+- Not supported
+
+---
+
+- `--global-name`
+- n/a
+- Not applicable, Bun does not support `iife` output at this time.
+
+---
+
+- `--ignore-annotations`
+- n/a
+- Not supported
+
+---
+
+- `--inject`
+- n/a
+- Not supported
+
+---
+
+- `--jsx`
+- `--jsx-runtime`
+- Supports `"automatic"` (uses `jsx` transform) and `"classic"` (uses `React.createElement`)
+
+---
+
+- `--jsx-dev`
+- n/a
+- Bun uses `jsxDEV` by default. Set `NODE_ENV=production` to use the production `jsx` transform instead.
+
+---
+
+- `--jsx-factory`
+- `--jsx-factory`
+
+---
+
+- `--jsx-fragment`
+- `--jsx-fragment`
+
+---
+
+- `--jsx-import-source`
+- `--jsx-import-source`
+
+---
+
+- `--jsx-side-effects`
+- n/a
+- JSX is always assumed to be side-effect-free.
+
+---
+
+- `--keep-names`
+- n/a
+- Not supported
+
+---
+
+- `--keyfile`
+- n/a
+- Not applicable
+
+---
+
+- `--legal-comments`
+- n/a
+- Not supported
+
+---
+
+- `--log-level`
+- n/a
+- Not supported. This can be set in `bunfig.toml` as `logLevel`.
+
+---
+
+- `--log-limit`
+- n/a
+- Not supported
+
+---
+
+- `--log-override:X=Y`
+- n/a
+- Not supported
+
+---
+
+- `--main-fields`
+- n/a
+- Not supported
+
+---
+
+- `--mangle-cache`
+- n/a
+- Not supported
+
+---
+
+- `--mangle-props`
+- n/a
+- Not supported
+
+---
+
+- `--mangle-quoted`
+- n/a
+- Not supported
+
+---
+
+- `--metafile`
+- `--manifest`
+
+---
+
+- `--minify-whitespace`
+- `--minify-whitespace`
+
+---
+
+- `--minify-identifiers`
+- `--minify-identifiers`
+
+---
+
+- `--minify-syntax`
+- `--minify-syntax`
+
+---
+
+- `--out-extension`
+- n/a
+- Not supported
+
+---
+
+- `--outbase`
+- `--root`
+- Not supported
+
+---
+
+- `--preserve-symlinks`
+- n/a
+- Not supported
+
+---
+
+- `--public-path`
+- `--public-path`
+
+---
+
+- `--pure`
+- n/a
+- Not supported
+
+---
+
+- `--reserve-props`
+- n/a
+- Not supported
+
+---
+
+- `--resolve-extensions`
+- n/a
+- Not supported
+
+---
+
+- `--servedir`
+- n/a
+- Not applicable
+
+---
+
+- `--source-root`
+- n/a
+- Not supported
+
+---
+
+- `--sourcefile`
+- n/a
+- Not supported. Bun does not support `stdin` input yet.
+
+---
+
+- `--sourcemap`
+- `--sourcemap`
+- No differences
+
+---
+
+- `--sources-content`
+- n/a
+- Not supported
+
+---
+
+- `--supported`
+- n/a
+- Not supported
+
+---
+
+- `--tree-shaking`
+- n/a
+- Always `true`
+
+---
+
+- `--tsconfig`
+- `--tsconfig-override`
+
+---
+
+- `--version`
+- n/a
+- Run `bun --version` to see the version of Bun.
+
+{% /table %}
+
+## JavaScript API
+
+{% table %}
+
+- `esbuild.build()`
+- `Bun.build()`
+
+---
+
+- `absWorkingDir`
+- n/a
+- Always set to `process.cwd()`
+
+---
+
+- `alias`
+- n/a
+- Not supported
+
+---
+
+- `allowOverwrite`
+- n/a
+- Always `true`
```

Comment:
this is true but we should really implement this to be default false. See:  https://discord.com/channels/1021987251930005594/1098562907941371914/1103115714006614058